spring-cloud-gcp-test/src/test/java/org/springframework/cloud/gcp/test/SpannerTemplateEmulatorTests.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.test;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Assumptions;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gcp.data.spanner.core.SpannerTemplate;
+import org.springframework.cloud.gcp.data.spanner.core.admin.SpannerDatabaseAdminTemplate;
+import org.springframework.cloud.gcp.data.spanner.core.admin.SpannerSchemaUtils;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.Column;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.Table;
+import org.springframework.cloud.gcp.test.spanner.SpannerEmulatorSpringConfiguration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * Spanner emulator tests.
+ *
+ * @author Dmitry Solomakha
+ */
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {SpannerEmulatorSpringConfiguration.class, SpannerTestConfiguration.class})
+public class SpannerTemplateEmulatorTests {
+	@Autowired
+	private SpannerTemplate spannerTemplate;
+
+	@Autowired
+	private SpannerDatabaseAdminTemplate spannerDatabaseAdminTemplate;
+
+	@Autowired
+	private SpannerSchemaUtils spannerSchemaUtils;
+
+	private static final Log LOGGER = LogFactory.getLog(SpannerTemplateEmulatorTests.class);
+
+	@BeforeClass
+	public static void checkToRun() {
+		Assumptions.assumeThat(System.getProperty("it.emulator"))
```

Comment:
It should be `-Dit.spanner-emulator`.
Also, can we update Travis CI config to actually run this test? What would it take?

Reply:
We already install gcloud, so installing emulators would be more of the same. Can be a follow up PR to update Travis config -- Dmitry wanted to add it right away, but I remember how annoying testing travis config is (multiple tries, the only way to test is by committing), so I asked for a separate minimal Travis config PR.